src/main/java/org/jabref/logic/importer/fileformat/ACMPortalParser.java

```diff
@@ -0,0 +1,241 @@
+            String jsonString = new String((stream.readAllBytes()), Charsets.UTF_8);
+
+            JsonElement jsonElement = JsonParser.parseString(jsonString);
+            if (jsonElement.isJsonObject()) {
+                JsonArray items = jsonElement.getAsJsonObject().getAsJsonArray("items");
+                for (JsonElement item : items) {
+                    for (Map.Entry<String, JsonElement> entry : item.getAsJsonObject().entrySet()) {
+                        bibEntries.add(parseBibEntry(entry.getValue().toString()));
+                    }
+                }
+            }
+        } catch (IOException | URISyntaxException e) {
+            throw new FetcherException("A network error occurred while fetching from ", e);
+        }
+
+        return bibEntries;
+    }
+
+    /**
+     * Constructing the query url for the doi
+     *
+     * @param doiList DOI List
+     * @return query URL
+     */
+    public URL getUrlFromDoiList(List<String> doiList) throws URISyntaxException, MalformedURLException {
+        URIBuilder uriBuilder = new URIBuilder(DOI_URL);
+        uriBuilder.addParameter("targetFile", "custom-bibtex");
+        uriBuilder.addParameter("format", "bibTex");
+        uriBuilder.addParameter("dois", String.join(",", doiList));
+        return uriBuilder.build().toURL();
+    }
+
+    private StandardEntryType typeStrToEnum(String typeStr) {
+        StandardEntryType type;
+        if ("PAPER_CONFERENCE".equals(typeStr)) {
+            type = StandardEntryType.Conference;
+        } else {
+            String upperUnderscoreTyeStr = CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.UPPER_CAMEL, typeStr);
+            type = Enums.getIfPresent(StandardEntryType.class, upperUnderscoreTyeStr).or(StandardEntryType.Article);
+        }
+        return type;
+    }
+
+    /**
+     * Parse BibEntry from query result xml
+     *
+     * @param jsonStr query result in JSON format
+     * @return BibEntry parsed from query result
+     */
+    public BibEntry parseBibEntry(String jsonStr) {
+        JsonObject jsonObject = JsonParser.parseString(jsonStr).getAsJsonObject();
+        BibEntry bibEntry = new BibEntry();
+        if (jsonObject.has("type")) {
+            bibEntry.setType(typeStrToEnum(jsonObject.get("type").getAsString()));
+        }
+
+        if (jsonObject.has("author")) {
+            JsonArray authors = jsonObject.getAsJsonArray("author");
+            StringBuilder authorStrBuilder = new StringBuilder();
+            int size = authors.size();
+            for (JsonElement author : authors) {
+                JsonObject authorJsonObject = author.getAsJsonObject();
+                authorStrBuilder.append(authorJsonObject.get("given").getAsString()).append(" ")
+                        .append(authorJsonObject.get("family").getAsString());
+                --size;
+                if (size > 0) {
+                    authorStrBuilder.append(" and ");
+                }
```

Comment:
Please use the magic of Java 8's [StringJoiner](https://docs.oracle.com/javase/8/docs/api/java/util/StringJoiner.html)

```suggestion
                authors.append(
                  authorJsonObject.get("given") + " " + authorJsonObject.get("family")
                );
```